components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/broker_agencies/broker_agency_staff_roles_controller.rb

```diff
@@ -1,31 +1,38 @@
+# frozen_string_literal: true
+
 module BenefitSponsors
   module Profiles
     module BrokerAgencies
+      # controller that manages adding, approving and removing of staff agency roles to a broker agency profile
       class BrokerAgencyStaffRolesController < ::BenefitSponsors::ApplicationController
 
         def new
+          profile = find_broker_agency_profile
+          authorize profile
+
           @staff = BenefitSponsors::Organizations::OrganizationForms::StaffRoleForm.for_new
           set_ie_flash_by_announcement
 
           respond_to do |format|
-            format.html { render 'new', layout: false} if params[:profile_type]
-            format.js  { render 'new'}
+            # previously rendered an HTML form which was unauthorized --> only JS responses should be allowed from this endpoint
+            format.js  { render 'new' }
           end
         end
 
         def create
```

Comment:
# The endpoint is used to create applications to existing broker agencies and can be sent by anyone
# This endpoint is unauthorized by design